Examples/Jax/jax_example_mps.py

```diff
@@ -9,12 +9,12 @@
 g = nk.graph.Hypercube(length=L, n_dim=1, pbc=True)
 
 # Hilbert space of spins on the graph
-hi = nk.hilbert.PySpin(graph=g, s=0.5)
+hi = nk.hilbert.Spin(graph=g, s=0.5)
 
 ha = nk.operator.Ising(h=1.0, hilbert=hi)
 
 ma = nk.machine.JaxMpsPeriodic(hi, bond_dim=4, diag=False, symperiod=-1, dtype=complex)
-ma.init_random_parameters(seed=1232)
+ma.jax_init_parameters(seed=1232)
```

Comment:
is there a reason you want to call this function instead of the more general `init_random_parameters`? I yesterday fixed the fact that it was not setting up the imaginary part of the components...

Both should be ok, as you specified a custom init for complex values, but I'd like to understand better the use case here

Reply:
I wrote a custom init function in the `MpsPeriodicLayer` which initializes every tensor as a unit matrix with (Gaussian) random numbers added to it.  This initialization leads to faster convergence of the wave function.  The `init_random_parameters` initializes all the weights using a normal distribution centered around zero, which leads to slower convergence for this MPS wave function.

Reply:
@gcarleo I wonder if in this case it would not be better to override `init_random_parameters` in the jax machine...

Reply:
yes, this is a broader issue, but I would find more consistent if init_random_parameters would do what the random initializer function of the layers wants to do... The issue is (for Tom) is that all default initializers in jax are for real parameters, so there is no way to have a default behavior where you get complex parameters if you want to. The solution is to do what you did and write another initializer... 

I think that maybe the "right" thing might be to just have init_random call the jax random initializers and then we should just check if the returned parameters are consistent with the dtype specified by the user. If they are not, we raise an exception?

Reply:
To make this more consistent with other machines, I could implement the identical part of the matrices in the `apply_fun`, so that all the parameters of this machine can also be initialized with values from a normal distribution centered around zero.  This would change the gradients a little bit, so I would have to check what the influence of this is in terms of convergence, but I expect the difference to be not that large.  Implemented like that, using the `init_random_parameters` would be suitable for this machine too.  Let me know what you think fits best for this machine.

Reply:
don't worry. I wouldn't go that far.
we should address this thing more in general

Reply:
but just to understand, if one calls init_random_parameters here instead of jax_init_parameters, does it happen that the tensors are initialized correctly with the gaussian weights?

Reply:
yes, the tensors are initialized and optimized correctly.  The problem is that this initialization leads to exploding/vanishing gradients depending on the spread of the Gaussian.  Using the initialization with the added unit matrix diminishes this problem.

Reply:
OK thanks! I think this is all we need for the parameter initizializations then...